### ISSUE TITLE
Fetching associations does an extra call with last version

### ISSUE THREAD

**Issue by charlesdemers** [charlieridley/emu#24]
Hi Charlie,

I thought I’d follow up [our discussion](http://discuss.emberjs.com/t/my-alternative-to-ember-data/916) here.

I pulled your last version and tested it with my example (it was working with the previous version):

``` js
App.Organization = Emu.Model.extend({
  name: Emu.field('string'),
  projects: Emu.field('App.Project', {collection: true, lazy: true})
});

App.Project = Emu.Model.extend({
  name: Emu.field('string'),
  organization: Emu.field('App.Organization')
});

App.Organization.find(1).get('projects');
```

Now it works but here are the calls that it does:

```
GET organizations/1
GET organizations/1/projects
GET undefined/projects
```

Do you have any idea why it would make the 3rd call?

Thanks!

**Comment by charlieridley** [comment]
Hi Charles.

Can't seem to replicate that. I've create a JSBin:

http://jsbin.com/odixan/1

When I open up the chrome console and run App.Organization.find(1).get('projects') I get this ouput:

```
DEBUG: ------------------------------- ember.js:339
DEBUG: Ember.VERSION : 1.0.0-rc.1 ember.js:339
DEBUG: Handlebars.VERSION : 1.0.0-rc.3 ember.js:339
DEBUG: jQuery.VERSION : 1.9.1 ember.js:339
DEBUG: ------------------------------- ember.js:339
App.Organization.find(1).get('projects');
Class {parent: Class, type: function, store: Class, createRecord: function, find: function…}
GET http://jsbin.com/odixan/organization/1 404 (Not Found) jquery-1.9.1.min.js:5
XHR finished loading: "http://jsbin.com/odixan/organization/1". jquery-1.9.1.min.js:5
GET http://jsbin.com/odixan/organization/1/project 404 (Not Found) jquery-1.9.1.min.js:5
XHR finished loading: "http://jsbin.com/odixan/organization/1/project". jquery-1.9.1.min.js:5

```

I would image that it might make a 3rd call because of my buggy code! Could you replicate the issue with a jsbin/jsfiddle and then I'll look into it.

Thanks

**Comment by charlesdemers** [comment]
I was able to reproduce the bug.

I trapped calls to `$.ajax` to log the urls called so if you open your devtools you should see the `undefinded/project` calls.

[http://jsbin.com/ewabes/7/edit](http://jsbin.com/ewabes/7/edit)

**Comment by charlieridley** [comment]
Ok thanks can see the issue now, will investigate

**Comment by charlieridley** [comment]
I fixed this issue, this integration test verifies the fix:

https://github.com/charlieridley/emu/blob/master/specs/emu/integration/lazy_loading_collection_tests.js.coffee#L39

**Comment by charlesdemers** [comment]
Hey thanks a lot, your fix works perfectly but I found another issue with bindings firing before the parent is ready... sorry :)

This time it’s when you create a record on a collection. As soon as it’s added to the collection, the bindings fire even if the record hasn’t been saved.

I made a JSBin to show you, you just have to click on the "Create task" button and watch your console.

[http://jsbin.com/ewabes/16/edit](http://jsbin.com/ewabes/16/edit)

Again, thanks this is very appreciated
